test/releasers/php.ts

```diff
@@ -0,0 +1,139 @@
+// limitations under the License.
+
+import {describe, it, afterEach} from 'mocha';
+import * as nock from 'nock';
+nock.disableNetConnect();
+
+import {PHP} from '../../src/releasers/php';
+import * as sinon from 'sinon';
+import {expect} from 'chai';
+import {buildGitHubFileRaw} from './utils';
+import {readFileSync} from 'fs';
+import {resolve} from 'path';
+import {GitHub} from '../../src/github';
+import {stubSuggesterWithSnapshot} from '../helpers';
+
+const sandbox = sinon.createSandbox();
+
+const fixturesPath = './test/fixtures';
+
+describe('PHP', () => {
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it('generates CHANGELOG and aborts if duplicate', async function () {
+    const releasePR = new PHP({
+      github: new GitHub({owner: 'googleapis', repo: 'release-please'}),
+      packageName: 'yoshi-php',
+    });
+
+    sandbox
+      .stub(releasePR.gh, 'getDefaultBranch')
+      .returns(Promise.resolve('master'));
+
+    // No open release PRs, so create a new release PR
+    sandbox
+      .stub(releasePR.gh, 'findOpenReleasePRs')
+      .returns(Promise.resolve([]));
+
+    // Indicates that there are no PRs currently waiting to be released:
+    sandbox
+      .stub(releasePR.gh, 'findMergedReleasePR')
+      .returns(Promise.resolve(undefined));
+
+    sandbox.stub(releasePR, 'latestTag').returns(
+      Promise.resolve({
+        name: 'v0.20.3',
+        sha: 'bf69d0f204474b88b3f8b5a72a392129d16a3929',
+        version: '0.20.3',
+      })
+    );
+
+    const graphql = JSON.parse(
+      readFileSync(
+        resolve(fixturesPath, 'commits-yoshi-php-monorepo.json'),
+        'utf8'
+      )
+    );
+    const req = nock('https://api.github.com')
+      // now we fetch the commits via the graphql API;
+      // note they will be truncated to just before the tag's sha.
+      .post('/graphql', () => {
+        return true;
+      })
+      .reply(200, {
+        data: graphql,
+      });
+
+    const getFileContentsStub = sandbox.stub(
+      releasePR.gh,
+      'getFileContentsOnBranch'
+    );
+    getFileContentsStub
+      .withArgs('AutoMl/composer.json', 'master')
+      .resolves(buildGitHubFileRaw('{"name": "automl"}'));
+    getFileContentsStub
+      .withArgs('AutoMl/VERSION', 'master')
+      .resolves(buildGitHubFileRaw('1.8.3'));
+    getFileContentsStub
+      .withArgs('Datastore/composer.json', 'master')
+      .resolves(buildGitHubFileRaw('{"name": "datastore"}'));
+    getFileContentsStub
+      .withArgs('Datastore/VERSION', 'master')
+      .resolves(buildGitHubFileRaw('2.0.0'));
+    getFileContentsStub
+      .withArgs('PubSub/composer.json', 'master')
+      .resolves(buildGitHubFileRaw('{"name": "pubsub"}'));
+    getFileContentsStub
+      .withArgs('PubSub/VERSION', 'master')
+      .resolves(buildGitHubFileRaw('1.0.1'));
+    getFileContentsStub
+      .withArgs('Speech/composer.json', 'master')
+      .resolves(buildGitHubFileRaw('{"name": "speech"}'));
+    getFileContentsStub
```

Comment:
It seems like we're recreating my team's use-case in this test, which is a PHP library with many composer files in it.

Would it make sense for us to have a more concise unit test, that just updates a root level composer file?

Reply:
You are absolutely right. I use copy/paste and refactor existing tests for my case. I agree with you, so I removed a lot of lines that duplicate test case with yoshi style.